app/src/main/java/com/keylesspalace/tusky/components/report/ReportActivity.kt

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2019 Joel Pyska
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.report
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.Spanned
+import android.view.MenuItem
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.keylesspalace.tusky.BaseActivity
+import com.keylesspalace.tusky.BottomSheetActivity
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.components.report.adapter.ReportPagerAdapter
+import com.keylesspalace.tusky.di.ViewModelFactory
+import com.keylesspalace.tusky.util.HtmlUtils
+import dagger.android.AndroidInjector
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.support.HasSupportFragmentInjector
+import kotlinx.android.synthetic.main.activity_report.*
+import kotlinx.android.synthetic.main.toolbar_basic.*
+import javax.inject.Inject
+
+class ReportActivity : BottomSheetActivity(), HasSupportFragmentInjector {
+
+    @Inject
+    lateinit var dispatchingFragmentInjector: DispatchingAndroidInjector<Fragment>
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    private lateinit var viewModel: ReportViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)[ReportViewModel::class.java]
+        val accountId = intent?.getStringExtra(ACCOUNT_ID)
+        val accountUserName = intent?.getStringExtra(ACCOUNT_USERNAME)
+        if (accountId.isNullOrBlank() || accountUserName.isNullOrBlank()) {
+            throw IllegalStateException("accountId ($accountId) or accountUserName ($accountUserName) is null")
+        }
+
+        viewModel.init(accountId, accountUserName,
+                intent?.getStringExtra(STATUS_ID), intent?.getStringExtra(STATUS_CONTENT))
+
+
+        setContentView(R.layout.activity_report)
+
+        setSupportActionBar(toolbar)
+
+        supportActionBar?.apply {
+            title = getString(R.string.report_username_format, viewModel.accountUserName)
+            setDisplayHomeAsUpEnabled(true)
+            setDisplayShowHomeEnabled(true)
+            setHomeAsUpIndicator(R.drawable.ic_close_24dp)
```

Comment:
You need to tint this drawable before using it otherwise it looks bad on the white theme. See here how its done in `ComposeActivity`: https://github.com/tuskyapp/Tusky/blob/master/app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java#L282

Reply:
Done